openmrs-client/src/main/java/org/openmrs/mobile/activities/patientdashboard/charts/PatientChartsFragment.java

```diff
@@ -71,16 +76,45 @@ public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup c
         setEmptyListVisibility(false);
 
         mListView.setOnItemClickListener((parent, view, position, id) -> {
-            Intent intent = new Intent(getActivity(), ChartsViewActivity.class);
-            Bundle mBundle = new Bundle();
+
             String vitalName = chartsListAdapter.getItem(position);
             try {
-                mBundle.putString("vitalName", observationList.getJSONObject(vitalName).toString());
+                JSONObject chartData = observationList.getJSONObject(vitalName);
+                Iterator<String> dates = chartData.keys();
+                ArrayList<String> dateList = Lists.newArrayList(dates);
+                Boolean validChartData = true;
+
+                if(dateList.size()==0)
+                    ToastUtil.showShortToast(getContext(), ToastUtil.ToastType.ERROR, getString(R.string.data_not_available_for_this_field));
+                else {
+                    JSONArray dataArray = chartData.getJSONArray(dateList.get(0));
+                    String entry = (String) dataArray.get(0);
+                    try {
+                        Float entryValue = Float.parseFloat(entry);
+                    } catch (NumberFormatException e) {
+                        validChartData = false;
+                        ToastUtil.showShortToast(getContext(), ToastUtil.ToastType.ERROR, getString(R.string.data_type_not_available_for_this_field));
+                    }
+                }
+
+                if(dateList.size()!=0 && validChartData) {
+                    Intent intent = new Intent(getActivity(), ChartsViewActivity.class);
+                    Bundle mBundle = new Bundle();
+                    try {
+                        mBundle.putString("vitalName", observationList.getJSONObject(vitalName).toString());
```

Comment:
`chartData` is defined as `observationList.getJSONObject(vitalName)`, right?
```suggestion
                        mBundle.putString("vitalName", chartData.toString());
```